apollo-federation/src/composition/satisfiability/validation_state.rs

```diff
@@ -298,8 +629,9 @@ impl ValidationState {
         if !context.is_shareable(field_definition_position)? {
             return Ok(Some(updated_state));
         }
-        let filtered_paths_count = updated_state
-            .subgraph_paths
+
+        let all_subgraph_paths: Vec<_> = updated_state.subgraph_path_infos.iter().collect();
```

Comment:
No need to collect at all. We can just call `iter()` directly in the line below (and another line down below).